hypha/static_src/src/app/src/redux/selectors/submissions.js

```diff
@@ -75,6 +75,27 @@ const getSubmissionReminders = createSelector(getCurrentSubmission, submission =
     return submission && submission.reminders || []
 })
 
+const getSubmissionMetaTerms = createSelector(
+    [getCurrentSubmission], submission => {
+        let metaTerms = []
+        if(submission && "metaTerms" in submission){
+            submission.metaTerms.map(metaTerm => {
+                const existingMetaTermIndex = metaTerms.findIndex(mt => mt.parentId == metaTerm.parentId);
+                if (existingMetaTermIndex != -1) {
+                    metaTerms[existingMetaTermIndex].children.push({'id': metaTerm.id, 'name': metaTerm.name})
```

Comment:
💭  We should consider using lodash in future to simplify array operations like grouping by a particular key. That could also be used for other cases to make the code more readable.

Reply:
which lodash function would you recommend for this?